src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -116,15 +182,65 @@ public async Task<IActionResult> GetInsights(string subscriptionId, string resou
             return await base.GetInsights(app, pesId, supportTopicId, startTime, endTime, timeGrain);
         }
 
+        /// <summary>
+        /// Publish detector package.
+        /// </summary>
+        /// <param name="pkg">The package.</param>
+        /// <returns>Task for publishing package.</returns>
         [HttpPost(UriElements.Publish)]
-        public async Task<IActionResult> PublishDetector(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DetectorPackage pkg)
+        public async Task<IActionResult> PublishDetectorPackage([FromBody] DetectorPackage pkg)
         {
-            return await base.PublishDetector(pkg);
+            return await base.PublishPackage(pkg);
         }
 
-        private bool IsPostBodyMissing(DiagnosticSiteData postBody)
+        /// <summary>
+        /// Publish gist package.
+        /// </summary>
+        /// <param name="pkg">The package.</param>
+        /// <returns>Task for publishing package.</returns>
+        [HttpPost(UriElements.PublishGist)]
+        public async Task<IActionResult> PublishGistPackage([FromBody] GistPackage pkg)
```

Comment:
Same here. You will need subscriptionId, resourceGroupName, and siteName because of the Route[SiteResource] at the controller level